src/StlViewer/SceneSetup.tsx

```diff
@@ -46,6 +47,16 @@ export interface SceneSetupProps {
   onFinishLoading?: (ev: ModelDimensions) => any
   modelProps?: ModelProps
   floorProps?: FloorProps
+  subdivide?: number
+  subdivideProps?: SubdivideProps
+}
+
+export interface SubdivideProps {
+  split: boolean
+  uvSmooth: boolean
+  preserveEdges: boolean
+  flatOnly: boolean
+  maxTriangles: number
 }
```

Comment:
I would expect `subdivideProps` to gather all the subdivision-related properties, as it is the user-facing API, it looks a bit more concise:
```suggestion
  subdivideProps?: Partial<SubdivideProps>
}

export interface SubdivideProps {
  subdivisions: number
  split: boolean
  uvSmooth: boolean
  preserveEdges: boolean
  flatOnly: boolean
  maxTriangles: number
}
```
Additionally, how about not requiring all the `SubdivideProps`, and instead, let the users just pass a partial subset of the `SubdivideProps` they need to change, providing reasonable defaults to the ones not set?

A user could provide something like this:
```tsx
<StlViewer subdivideProps={{ subdivisions: 2, preserveEdges: false }} />
```
And all the other fields could have the following default values:
```ts
const DEFAULT_SUBDIVIDE_PROPS:  SubdivideProps = {
  subdivisions: 0,
  split: true,
  uvSmooth: false,
  preserveEdges: false,
  flatOnly: false,
  maxTriangles: Infinity
}
```